File: java/socketd/src/main/java/org/noear/socketd/transport/client/ClientHeartbeatHandler.java
package org.noear.socketd.transport.client;

import org.noear.socketd.transport.core.Session;

/**
 * 客户端心跳处理器
 *
 * @author noear
 * @since 2.0
 */
@FunctionalInterface
public interface ClientHeartbeatHandler {
    /**
     * 心跳处理
     */
    void clientHeartbeat(Session session) throws Exception;
}

File: javascript/socketd/src/socketd/transport/client/ClientHeartbeatHandler.ts
import {Session} from "../core/Session";
import {IoConsumer} from "../core/Typealias";


export interface ClientHeartbeatHandler {
    clientHeartbeat(session:Session);
}

export class ClientHeartbeatHandlerDefault implements ClientHeartbeatHandler {
    private _heartbeatHandler: IoConsumer<Session> | null;

    constructor(heartbeatHandler: IoConsumer<Session> | null) {
        this._heartbeatHandler = heartbeatHandler;

    }

    clientHeartbeat(session: Session) {
        if (this._heartbeatHandler) {
            this._heartbeatHandler!(session);
        } else {
            session.sendPing();
        }
    }
}